### PR TITLE
feat: implement seed method on InMemoryDBService

### PR DESCRIPTION
**Pull request by wesleygrimes**
Add need seed method on InMemoryDBService with ability to set amount and record factory for
generation.

impl #7

### CODE REVIEW

**Comment by wesleygrimes** [comment]
@nestjs-addons/collaborators  I think this is ready for review, one question? 

Would it add value to give developers "random helpers" or in the record factory generator?

```typescript
const recordFactory = 
  (index: number, randomString: string, randomNumber: number, randomDate: Date) 
    => ({ myProp: randomString, myOtherProp: `${index}-test-value` });
```

File: lib/services/in-memory-db.service.ts [hunk 1]
```diff
@@ -296,6 +296,25 @@ export class InMemoryDBService<T extends InMemoryDBEntity> {
...
+   * service.seed((i) => { myProp: i}, 100);
+   * ```
+   *
+   * @param recordFactory a factory method to call when generating the random record.
+   * @param amount the amount of records to generate, defaults to 10.
+   */
+  public seed(recordFactory: (index: number) => Partial<T>, amount = 10) {
+    amount = amount === null ? 10 : amount;
+
+    for (let i = 0; i < amount; i++) {
```
Comment (cmwhited) [inline lib/services/in-memory-db.service.ts]: Could we utilize the `createMany` method here instead of this `for` loop?

**Review by cmwhited — DISMISSED** [review]
Looks good! I put a comment about using the `createMany` method in the seed, since that was part of our initial discussion to add that method.

File: lib/services/in-memory-db.service.ts (hunk 1 above)
Comment (wesleygrimes) [reply lib/services/in-memory-db.service.ts]: Good point! I will switch it out.

**Review by cmwhited — DISMISSED** [review]

**Review by cmwhited — APPROVED** [review]

**Pull request merged**